simulariumio/filters/translate_filter.py

```diff
@@ -67,24 +66,29 @@ def apply(self, data: TrajectoryData) -> TrajectoryData:
                     ]
                 else:
                     translation = self.default_translation
-                n_subpoints = int(data.agent_data.n_subpoints[time_index][agent_index])
-                if n_subpoints > 0:
-                    for subpoint_index in range(
-                        int(data.agent_data.n_subpoints[time_index][agent_index])
-                    ):
-                        for dim in range(3):
-                            subpoints[time_index][agent_index][subpoint_index][dim] = (
-                                data.agent_data.subpoints[time_index][agent_index][
-                                    subpoint_index
-                                ][dim]
-                                + translation[dim]
-                            )
-                else:
-                    for dim in range(3):
-                        positions[time_index][agent_index][dim] = (
-                            data.agent_data.positions[time_index][agent_index][dim]
-                            + translation[dim]
-                        )
-        data.agent_data.positions = positions
-        data.agent_data.subpoints = subpoints
+                # apply translation
+                display_type = data.agent_data.display_type_for_agent(
+                    time_index, agent_index
+                )
+                translate_subpoints = (
+                    max_subpoints > 0 and display_type != DISPLAY_TYPE.SPHERE_GROUP
+                )
+                if translate_subpoints:
+                    sp_items = self.get_items_from_subpoints(
+                        data.agent_data, time_index, agent_index
+                    )
+                    if sp_items is None:
+                        translate_subpoints = False
+                    else:
+                        # translate subpoints for fibers
```

Comment:
I think we ideally want fiber points to be relative to the fiber agent's position, but the viewer wasn't rendering it that way, at least partially because of a bug. I made an issue for it, I think the viewer can handle that now? https://github.com/simularium/simulariumio/issues/109